src/compiler/sys.ts

```diff
@@ -1353,23 +1353,32 @@ namespace ts {
             function getAccessibleFileSystemEntries(path: string): FileSystemEntries {
                 perfLogger.logEvent("ReadDir: " + (path || "."));
                 try {
-                    const entries = _fs.readdirSync(path || ".").sort();
+                    const entries = _fs.readdirSync(path || ".", { withFileTypes: true });

```

Comment:
@amcasey does this need node version check that you had in your PR ?

Reply:
It’s not needed as written; old Node just ignores the unknown option.

```console
$ docker run --rm -it node:8-alpine -p 'fs.readdirSync("/", { withFileTypes: true })'
[ '.dockerenv',
  'bin',
  'dev',
  'etc',
  'home',
  'lib',
  'media',
  'mnt',
  'opt',
  'proc',
  'root',
  'run',
  'sbin',
  'srv',
  'sys',
  'tmp',
  'usr',
  'var' ]
$ docker run --rm -it node:10-alpine -p 'fs.readdirSync("/", { withFileTypes: true })'
[ Dirent { name: '.dockerenv', [Symbol(type)]: 1 },
  Dirent { name: 'bin', [Symbol(type)]: 2 },
  Dirent { name: 'dev', [Symbol(type)]: 2 },
  Dirent { name: 'etc', [Symbol(type)]: 2 },
  Dirent { name: 'home', [Symbol(type)]: 2 },
  Dirent { name: 'lib', [Symbol(type)]: 2 },
  Dirent { name: 'media', [Symbol(type)]: 2 },
  Dirent { name: 'mnt', [Symbol(type)]: 2 },
  Dirent { name: 'opt', [Symbol(type)]: 2 },
  Dirent { name: 'proc', [Symbol(type)]: 2 },
  Dirent { name: 'root', [Symbol(type)]: 2 },
  Dirent { name: 'run', [Symbol(type)]: 2 },
  Dirent { name: 'sbin', [Symbol(type)]: 2 },
  Dirent { name: 'srv', [Symbol(type)]: 2 },
  Dirent { name: 'sys', [Symbol(type)]: 2 },
  Dirent { name: 'tmp', [Symbol(type)]: 2 },
  Dirent { name: 'usr', [Symbol(type)]: 2 },
  Dirent { name: 'var', [Symbol(type)]: 2 } ]
```